src/components/common/Tour.js

```diff
@@ -0,0 +1,256 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import JoyRide, { ACTIONS, EVENTS, STATUS } from 'react-joyride';
+import { withRouter } from 'react-router';
+import { connect } from 'react-redux';
+import { withTranslation } from 'react-i18next';
+import {
+  completeTour,
+  initializeTour,
+  navigateStopTour,
+  nextStepTour,
+  resetTour,
+  startTour,
+} from '../../actions';
+import { HOME_PATH, SETTINGS_PATH, VISIT_PATH } from '../../config/paths';
+import {
+  AUTHENTICATED,
+  THEME_COLORS,
+  USER_MODES,
+} from '../../config/constants';
+import {
+  SPACE_PREVIEW_ICON_CLASS,
+  VISIT_SPACE_INPUT,
+} from '../../config/selectors';
+import { waitForElement } from '../../utils/elements';
+import {
+  SETTINGS_TOUR_STEPS,
+  tours,
+  VISIT_SPACE_TOUR_STEPS,
+} from '../../config/tours';
+
+// eslint-disable-next-line react/prefer-stateless-function
+export class Tour extends Component {
+  static propTypes = {
+    dispatchNextStep: PropTypes.func.isRequired,
+    dispatchStartTour: PropTypes.func.isRequired,
+    dispatchNavigateAndStopTour: PropTypes.func.isRequired,
+    dispatchTourCompleted: PropTypes.func.isRequired,
+    dispatchResetTour: PropTypes.func.isRequired,
+    dispatchInitializeTour: PropTypes.func.isRequired,
+    tourKey: PropTypes.string.isRequired,
+    currentTour: PropTypes.oneOf(tours).isRequired,
+    run: PropTypes.bool.isRequired,
+    continuous: PropTypes.bool.isRequired,
+    stepIndex: PropTypes.number.isRequired,
+    steps: PropTypes.arrayOf(
+      PropTypes.shape({
+        target: PropTypes.string.isRequired,
+        title: PropTypes.string,
+        content: PropTypes.string.isRequired,
+        disableBeacon: PropTypes.bool,
+      })
+    ).isRequired,
+    signInActivity: PropTypes.bool.isRequired,
+    visitSpaceTourRan: PropTypes.bool.isRequired,
+    settingsTourRan: PropTypes.bool.isRequired,
+    authenticated: PropTypes.bool.isRequired,
+    location: PropTypes.shape({
+      pathname: PropTypes.string,
+    }).isRequired,
+    history: PropTypes.shape({
+      replace: PropTypes.func.isRequired,
+      push: PropTypes.func.isRequired,
+    }).isRequired,
+  };
+
+  componentDidUpdate(prevProps) {
+    const {
+      signInActivity,
+      authenticated,
+      visitSpaceTourRan,
+      settingsTourRan,
+      dispatchResetTour,
+      dispatchInitializeTour,
+      location: { pathname },
+      currentTour,
+      run,
+    } = this.props;
+    const { signInActivity: prevSignInActivity } = prevProps;
+
+    if (
+      pathname === HOME_PATH &&
+      prevSignInActivity &&
+      !signInActivity &&
+      authenticated &&
+      !visitSpaceTourRan
+    ) {
+      setTimeout(
```

Comment:
Can you add a comment why a `setTimeout` is necessary here?

Reply:
Here without a setTimout is necessary to wait for the element to render. Without having it and running the tour leads to an error. I tried avoiding the setTimeouts as much as possible especially with the use of waitForElement. But in that case it is the first step.

Reply:
The checking on the change in the activity ensures it doesn't run multiple times.